__tests__/CommentsTouchBox-test.js

```diff
@@ -0,0 +1,104 @@
+import 'react-native';
+import React from 'react';
+import CommentsTouchBox from '@screens/app/Components/CommentsTouchBox/CommentsTouchBox';
+import renderer from 'react-test-renderer';
+import {shallow} from 'enzyme';
+import checkPropTypes from 'check-prop-types';
+var assert = require('assert');
+
+test('like button snapshot test', () => {
+  const treeWithoutCommentsLength = renderer
+    .create(<CommentsTouchBox />)
+    .toJSON();
+  expect(treeWithoutCommentsLength).toMatchSnapshot();
+  const treeWithCommentLengthZero = renderer
+    .create(<CommentsTouchBox commentsLength={0} />)
+    .toJSON();
+  expect(treeWithCommentLengthZero).toMatchSnapshot();
+  const treeWithCommentLengthOne = renderer
+    .create(<CommentsTouchBox commentsLength={1} />)
+    .toJSON();
+  expect(treeWithCommentLengthOne).toMatchSnapshot();
+  const treeWithCommentLengthGreaterThanOne = renderer
+    .create(<CommentsTouchBox commentsLength={2} />)
+    .toJSON();
+  expect(treeWithCommentLengthGreaterThanOne).toMatchSnapshot();
+});
+
+function testPropTypesWithoutErrors(commentsLength) {
+  it(
+    'testing commentsLength values (No errors) where commentsLength is ' +
+      commentsLength,
+    () => {
+      const wrapper = shallow(
+        <CommentsTouchBox commentsLength={commentsLength} />,
+      );
+      const instance = wrapper.instance();
+      const commentsLengthProp = instance.props.commentsLength;
+      expect(commentsLength).toEqual(commentsLengthProp);
+    },
+  );
+}
+
+function testPropTypesWithErrors(commentLength, errorMsg, testDescription) {
+  it('Throws failed propType Error for ' + testDescription, () => {
+    let result = checkPropTypes(
+      CommentsTouchBox.propTypes,
+      {commentsLength: commentLength},
+      'prop',
+      CommentsTouchBox.commentsLength,
+    );
+    assert(result === errorMsg);
+  });
+}
+
+describe('testing commentsLength values', () => {
+  testPropTypesWithoutErrors(0);
```

Comment:
I think this should fail. We don't want '' as a comment on any post. In that case we would just delete it?

Reply:
CommentTouchBox does not show the comments, its just a box to redirect the user to the comments page. Maybe we can have it so it starts with 1?